programs/stake/src/stake_state.rs

```diff
@@ -842,6 +817,83 @@ impl<'a> StakeAccount for KeyedAccount<'a> {
     }
 }
 
+/// After calling `validate_delegate_amount()`, this struct contains calculated values that are used
+/// by the caller.
+struct ValidatedDelegateInfo {
+    stake_amount: u64,
+}
+
+/// Ensure the stake delegation amount is valid.  This checks that the account meets the minimum
+/// balance requirements to delegate stake.  If not, return an error.
+fn validate_delegate_amount(
+    account: &KeyedAccount,
+    meta: &Meta,
+) -> Result<ValidatedDelegateInfo, InstructionError> {
+    let stake_amount = account.lamports()?.saturating_sub(meta.rent_exempt_reserve); // can't stake the rent ;)
+    Ok(ValidatedDelegateInfo { stake_amount })
+}
```

Comment:
In a subsequent PR, this function will be expanded to enforce that the stake amount is at least the minimum stake delegation. But for now, to not change the current behavior, skip that check.